searx/engines/qwant.py

```diff
@@ -76,38 +105,74 @@ def request(query, params):
     if not query:
         return None
 
-    count = 10  # web: count must be equal to 10
+    q_locale = traits.get_region(params["searxng_locale"], default='en_US')
+
+    url = api_url + f'{qwant_categ}?'
+    args = {'q': query}
+    params['raise_for_httperror'] = False
+
+    if qwant_categ == 'web-lite':
+
+        url = web_lite_url + '?'
+        args['locale'] = q_locale.lower()
+        args['l'] = q_locale.split('_')[0]
+        args['s'] = params['safesearch']
+        args['p'] = params['pageno']
+
+        params['raise_for_httperror'] = True
+
+    elif qwant_categ == 'images':
 
-    if qwant_categ == 'images':
-        count = 50
-        offset = (params['pageno'] - 1) * count
+        args['locale'] = q_locale
+        args['safesearch'] = params['safesearch']
+
+        args['count'] = 50
+        offset = (params['pageno'] - 1) * args['count']
         # count + offset must be lower than 250
-        offset = min(offset, 199)
-    else:
-        offset = (params['pageno'] - 1) * count
-        # count + offset must be lower than 50
-        offset = min(offset, 40)
+        args['offset'] = min(offset, 199)
 
-    params['url'] = url.format(
-        keyword=qwant_categ,
-        query=urlencode({'q': query}),
-        offset=offset,
-        count=count,
-    )
+    else:  # web, news, videos
 
-    # add quant's locale
-    q_locale = traits.get_region(params["searxng_locale"], default='en_US')
-    params['url'] += '&locale=' + q_locale
+        args['locale'] = q_locale
+        args['safesearch'] = params['safesearch']
 
-    # add safesearch option
-    params['url'] += safe_search_map.get(params['safesearch'], '')
+        args['count'] = 10
+        offset = (params['pageno'] - 1) * args['count']
+        # count + offset must be lower than 50
+        args['offset'] = min(offset, 39)
+
+    params['url'] = url + urlencode(args)
 
-    params['raise_for_httperror'] = False
     return params
 
 
 def response(resp):
-    """Get response from Qwant's search request"""
+
+    if qwant_categ == 'web-lite':
+        return parse_web_lite(resp)
+    return parse_web_api
+
+
+def parse_web_lite(resp):
+    """Parse results from Qwant-Lite"""
+
+    results = []
+    dom = lxml.html.fromstring(resp.text)
+
+    for item in eval_xpath_list(dom, '//section/article'):
+        results.append(
+            {
+                'url': extract_text(item.xpath('./span')),
```

Comment:
Sometimes we use `eval_xpath(dom, selector)`, and sometimes we use `item.xpath(selector)`.

Referring to [utils.py](https://github.com/searxng/searxng/blob/7dfcc3386e6787fe2267ae667fefff51cc4286a6/searx/utils.py#L481), it seems like we should use `eval_xpath` instead because it's cached and hence provides a better performance?

Reply:
Oops you have found a point where I am often sloppy (will fix it) / to answer your question: yes, its better we use the xpath functions from the utils.py module. Theses functions will raise dedicated SearX-Xpath exceptions ..

https://github.com/searxng/searxng/blob/b4e0d2eedc94df2369a7661f9134fcfd8fd68af4/searx/exceptions.py#L111-L112

https://github.com/searxng/searxng/blob/b4e0d2eedc94df2369a7661f9134fcfd8fd68af4/searx/exceptions.py#L41-L42

Reply:
Alright, thanks for the explanation :+1:

Reply:
Fixed the issues / BTW I found another issue ..

```python
        # qwant-lite delivers only 5 pages maximum
        if params['pageno'] > 5:
            return
```